lib/bigintToString.ts

```diff
@@ -0,0 +1,14 @@
+// Recursive function for big int conversions to string for JSON serialisation.
+export function bigintToString(obj: any): any {
+  if (typeof obj === 'bigint') return obj.toString()
+
+  if (Array.isArray(obj)) return obj.map(bigintToString)
+
+  if (obj && typeof obj === 'object') {
```

Comment:
a correct check for object would be 
```
    obj !== null &&
    typeof obj === 'object' &&
    Object.getPrototypeOf(obj) === Object.prototype
```

for example dates and regexps are also objects, but you can't iterate over them, so Object.entries(obj) would fail for them